src/storage/btree.c

```diff
@@ -8258,9 +8258,10 @@ btree_repair_prev_link (THREAD_ENTRY * thread_p, OID * oid, BTID * index_btid, b
 	  break;
 	}
     }
+  /* if the break statement occurs due to an error, the lock acquired immediately before should be released. */
   if (!OID_ISNULL (&class_oid))
     {
-      lock_unlock_object (thread_p, &class_oid, oid_Root_class_oid, SCH_S_LOCK, true);
+      lock_unlock_object (thread_p, &class_oid, oid_Root_class_oid, IX_LOCK, true);
```

Comment:
> So requesting to release the lock different from the actual lock didn't make problems.

- What is the upside of using IX_LOCK instead of SCH_S_LOCK to unlock this? Please elaborate on it. 
- Is it locked with SCH_S_LOCK?

Reply:
No. It is locked with IX_LOCK, but lock_unlock_object requested SCH_S_LOCK.
so It should be fixed.

Reply:
Oh, I forgot to check the Jira description. Sorry.